.github/workflows/build.yml

```diff
@@ -0,0 +1,50 @@
+on: push
+name: Build
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: on
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/kubeflow/common
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+        path: ${{ env.GOPATH }}/src/github.com/kubeflow/common
```

Comment:
I can not remember the default behavior here. Without specifying the ref, does it by default 
1. fetch HEAD for `pull_request`? 
2. fetch master for `push`

Reply:
> The branch, tag or SHA to checkout. When checking out the repository that
> triggered a workflow, this defaults to the reference or SHA for that event.
> Otherwise, uses the default branch.

Reply:
So I guess it fetches HEAD of current branch in both of those events, at least it's also how I remember. I can double check though

Reply:
Ok yeah, as per docs:

1. [pull_request](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) -> Last merge commit on the GITHUB_REF branch (master)
2. [push](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push) -> Commit pushed